### PR TITLE
add fees support for mainnet transaction

### LINKED ISSUE #85: [FEATURE] Provide fee for transactions
Zwalletcli sends transactions with 0 fees, it won't work on mainnet
E. g.
``
		txn, err := zcncore.NewTransaction(statusBar, 0)
``

I propose to make optional key `--withFees = ` to provide fee and if not provided use `min_fee` as default

### PR DESCRIPTION
**Pull request by bisakhmondal**
A brief description of the changes in this PR:

closes: #85 

Incorporate
- `withFee` flag
- singleton `transactionFee` get method
- (TODO): in absence of withFee fallback to min_fee captured from blockchain

Tasks to complete before merging PR:
- [x]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zwalletcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- blobber:
- gosdk: https://github.com/0chain/gosdk/pull/623
- system_test:
- zboxcli:
- 0chain: https://github.com/0chain/0chain/pull/1753
- Other: ...

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -21,6 +21,11 @@ var cDir string
 var bSilent bool
 var nonce int64
 
+// MinTxFee sets the min tx fee that must be paid for a tx to occur.
+// making it public for the flexibility to set it through ldflags.
+var MinTxFee float64
```
Comment (dabasov) [inline cmd/root.go]: we should use int64 in SAS

File: cmd/root.go [hunk 2]
```diff
@@ -21,6 +21,11 @@ var cDir string
...
+var txFee float64
```
Comment (dabasov) [inline cmd/root.go]: the same int64 in SAS

**Review by dabasov — CHANGES_REQUESTED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by dabasov** [comment]
please fix tests @bisakhmondal

**Comment by bisakhmondal** [comment]
> please fix tests @bisakhmondal

On it.

**Comment by bisakhmondal** [comment]
> please fix tests @bisakhmondal

Hey @dabasov, I think we have to deploy the 0chain core to devnet for the estimate_tx_cost API
image

Any heads up on how to do that?

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee-integration  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by boddumanohar** [comment]
@dabasov are these changes good to merge?

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | chain-fee-remove-excess-faucet  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee-v2  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee-v2  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0dns  | staging  |

**Pull request merged**